Sonic/Player/Character/Speed/PlayerSpeedContext.h

```diff
@@ -2,8 +2,237 @@
+        bool eStateFlag_InvokeBarrier;
+        bool eStateFlag_InvokePreciousTime;
+        bool eStateFlag_InvokeHighSpeed;
+        bool eStateFlag_InvokeTimeBreak;
+        bool eStateFlag_InvokeSkateBoard;
+        bool eStateFlag_InvokeSuperSonic;
+        bool eStateFlag_InvokePtmRocket;
+        bool eStateFlag_InvokePtmSpike;
+        bool eStateFlag_InvokeUnbeaten;
+        bool eStateFlag_DoubleJumping;
+        bool eStateFlag_InvokeFixedMoveOn3D;
+        bool eStateFlag_KeepStateAfterChangeDimension;
+        bool eStateFlag_KeepPostureAfterChangeDimension;
+        bool eStateFlag_CalibrateFrontDir;
+        bool eStateFlag_EnableAnimationVelocityDirect;
+        bool eStateFlag_EnableGravityControl;
+        bool eStateFlag_EnableAirBoostOnGravityControl;
+        bool eStateFlag_EnableChaosEnergySetting;
+        bool eStateFlag_ChaosEnergySubstractDisabled;
+        bool eStateFlag_EnableAutoAim;
+        bool eStateFlag_ThroughGoalSignboard;
+        bool eStateFlag_AutoBoost;
+        bool eStateFlag_MaxSpeedLimited;
+        bool eStateFlag_Pause;
+        bool eStateFlag_SkateBoardSlowDown;
+        bool eStateFlag_ChangeCollisionLocked;
+        bool eStateFlag_ReadyGoOpened;
+        bool eStateFlag_SpikeSpin;
+        bool eStateFlag_ChangePostureInDeadAirEnabled;
+        bool eStateFlag_DisableGroundSmoke;
+        bool eStateFlag_Drifting;
+        bool eStateFlag_DriftingLowSpeed;
+        bool eStateFlag_DriftRight;
+        bool eStateFlag_Squat;
+        bool eStateFlag_LightSpeedDashReady;
+        bool eStateFlag_BoostKeep;
+        bool eStateFlag_DisableCrashWall;
+        bool eStateFlag_WallCorrection;
+        bool eStateFlag_Diving;
+        bool eStateFlag_DivingFloat;
+        bool eStateFlag_HipSliding;
+        bool eStateFlag_GrindFullBalance;
+        bool eStateFlag_GrindSideJump;
+        bool eStateFlag_Freeze;
+        bool eStateFlag_DisableBoost;
+        bool eStateFlag_NoSquatToSliding;
+        bool eStateFlag_IgnorePadInput;
+        bool eStateFlag_Rising;
+        bool eStateFlag_SpinChargeSliding;
+        bool eStateFlag_CombinationAttacking;
+        bool eStateFlag_SpinDash;
+        bool eStateFlag_LightAttack;
+        bool eStateFlag_AtomicSpin;
+        bool eStateFlag_Binding;
+        bool eStateFlag_ChgPlayerDisabled;
+        bool eStateFlag_EndReflection;
+        bool eStateFlag_CorrectOnPath;
+        bool eStateFlag_DebugDrawGroundHitPosition;
+        bool eStateFlag_DebugDrawVelocity;
+        bool eStateFlag_DebugDrawPath;
+        bool eStateFlag_DebugDrawSpikeInfo;
+        bool eStateFlag_DebugExceptionalMove;
+    };
+    // TODO: Further documentation on this.
+    // Its main use is getting state flags right now but it's not clear *WHAT* this is, or if anything else is useful.
+    struct SContainsStateFlags
+    {
+        char field_0;
+        char field_1;
+        char field_2;
+        char field_3;
+        eStateFlags* StateFlags;
```

Comment:
This looks like a hh::vector<uint8_t>? 
I'm assuming "SContainsStateFlags" is actually CStateFlags itself and it contains a hh::vector<uint8_t> for each StateFlag, hence the pointer to a separate memory block.

In that case, the eStateFlag should preferably be made into an enum type instead, and the flags should be accessed by index.